### PR TITLE
Fix login problem where an error is printed, despite login success, improve success/failure message, and don't print password.

### PR DESCRIPTION
**Pull request by kaladay**
The primary problem is that the token information may be undefined. The `config.set()` prints the observed error when this variable is undefined. Add wrapping conditional block to ensure that this error is not triggered on undefined or NULL.

Add additional sanity checks to make sure variable processing does not have undefined or NULL related errors.

It is very unclear on whether or not the login succeeded. Change the `resolve()` and `reject()` behavior to return a JSON object that has the status message in it.

For security reasons, do not console log the password. Put the user name in both the login and the failure messages.

### CODE REVIEW

**Review by rmathew1011 — APPROVED** [review]

File: src/service/okapi.service.ts [hunk 1]
```diff
@@ -27,18 +27,19 @@ class OkapiService extends RestService {
           'Accept': 'application/json'
         }
       }, (error: any, response: any, body: any) => {
-        if (response && response.statusCode >= 200 && response.statusCode <= 299) {
+        if (!!response && response?.statusCode >= 200 && response?.statusCode <= 299) {
```
Comment (wwelling) [inline src/service/okapi.service.ts:30]: Remove unnecessary conditions.

```suggestion
        if (!!response && response.statusCode >= 200 && response.statusCode <= 299) {
```

File: src/service/okapi.service.ts [hunk 2]
```diff
@@ -27,18 +27,19 @@ class OkapiService extends RestService {
...
           let accessToken;
           let refreshToken;
+          const headers = !!response?.headers ? response?.headers : undefined;
 
-          if (!!response.headers['set-cookie']) {
+          if (!!headers['set-cookie']) {
             const matchAccess = /folioAccessToken=([^;\s]*)/gi;
             const matchRefresh = /folioRefreshToken=([^;\s]*)/gi;
-            const cookies = !!response.headers['set-cookie'] ? response.headers['set-cookie'] : undefined;
+            const cookies = !!headers['set-cookie'] ? headers['set-cookie'] : undefined;
```
Comment (wwelling) [inline src/service/okapi.service.ts:38]: Again, I think we should consider whether it is a good practice to assign variables as undefined.

File: src/service/okapi.service.ts (hunk 1 above)
Comment (kaladay) [reply src/service/okapi.service.ts:30]: These are required as the existence of `response` does not mean `statusCode` is a present property.

Comment (wwelling) [reply src/service/okapi.service.ts:30]: This is still a copy and paste of an incorrect implementation that includes redundant conditions.

**Pull request merged**